### ISSUE TITLE
SA1101 false-positive report for generic local functions

### ISSUE THREAD

**Issue by kasthack** [DotNetAnalyzers/StyleCopAnalyzers#3012]
Reproduction:

````csharp
class A
{
    private void Reproduce()
    {
        Works();             //works just fine
        Broken<object>();    //false-positive SA1101 here

        void Works() { }
        void Broken<T>() { }
    }
}
````

**Comment by sharwell** [comment]
While adding a new test, I wasn't able to reproduce this in the latest builds. Do you know what version this was observed in?

**Comment by kasthack** [comment]
@sharwell Tested with `1.1.118`(latest) from nuget.

**Comment by vweijsters** [comment]
I cannot reproduce this in VS2017 (15.9.15) and VS2019 (16.2.3) with either 1.1.118 or 1.2.0-beta.66.

**Comment by kasthack** [comment]
My bad, the company's nuget proxy was messing with package versions and returning 1.0.0 instead of the requested ones. Haven't noticed that until I tried to reproduce the behavior on AppVeyor.

**Comment by julealgon** [comment]
Folks, I can still see this behavior in .Net Core 3.1, latest version of StyleCop.Analyzers. Minimal repro:

```csharp
        private sealed class SomeClass
        {
            public SomeClass()
            {
                void SomeLocalFunction<T>()
                {
                }

                _ = SomeLocalFunction<object>();
            }
        }
```

<img width="458" alt="stylecop-localfunc" src="https://user-images.githubusercontent.com/461579/71896918-c7004c00-3133-11ea-87da-ad71bda32b69.png">

Can we please reopen this?

**Comment by sharwell** [comment]
@julealgon This is fixed in 1.2.0-beta.113
